### PR TITLE
Fixing up hydra code, now provides url on unpublish

### PR DESCRIPTION
**Pull request by benradey**

### CODE REVIEW

**Comment by xdavidson** [comment]
When attempting to edit metadata so that I can publish a module, after select a product, I am presented with a black screen on pantheon ui with the following exception in karaf log:

service: Uncaught Throwable
java.lang.NullPointerException: null
	at org.eclipse.jetty.server.ResponseWriter.write(ResponseWriter.java:246) ~[!/:9.4.22.v20191022]
	at org.apache.sling.engine.impl.SlingHttpServletResponseImpl$1.write(SlingHttpServletResponseImpl.java:399) ~[!/:?]
	at com.redhat.pantheon.servlet.DocumentCustomerPortalUrlServlet.doGet(DocumentCustomerPortalUrlServlet.java:46) ~[?:?]

**Comment by benradey** [comment]
@xdavidson I couldn't reproduce locally but I think I know what you ran into. Can you see if the latest commit fixed it?

**Comment by xdavidson** [comment]
ben,

Here are a few things I found in this branch:
  1.I can publish/unpublish existing modules/assembly.  view_uri in the unpublish message has the wrong locale value
eg.  https://access.devgssci.devlab.phx1.redhat.com/documentation/en_US/red_hat_enterprise_linux/8-3/topic/5fd1a80c-2d41-4ec3-b440-deba234d742c"
It should be:
https://access.devgssci.devlab.phx1.redhat.com/documentation/en-us/red_hat_enterprise_linux/8-3/topic/5fd1a80c-2d41-4ec3-b440-deba234d742c"
  2. same locale issue on module/assembly variant api, the locale in the view_uri has the wrong value
 3. same locale issue on the UI for View on Customer Portal and Copy Permanent URL

  4. After uploading new module/assembly, on the display page when I attempt to Edit Metadata, Select product will trigger a black screen. I see the following exception in the browser console:
react-dom.production.min.js:187 TypeError: Cannot read property 'map' of undefined
    at Versions.render (versions.tsx:348)
    at Mh (react-dom.production.min.js:173)
    at Lh (react-dom.production.min.js:172)
    at mj (react-dom.production.min.js:247)
    at jj (react-dom.production.min.js:216)
    at Z (react-dom.production.min.js:210)
    at $i (react-dom.production.min.js:206)
    at react-dom.production.min.js:114
    at exports.unstable_runWithPriority (scheduler.production.min.js:20)
    at vf (react-dom.production.min.js:113)
di @ react-dom.production.min.js:187
versions.tsx:348 Uncaught (in promise) TypeError: Cannot read property 'map' of undefined
    at Versions.render (versions.tsx:348)
    at Mh (react-dom.production.min.js:173)
    at Lh (react-dom.production.min.js:172)
    at mj (react-dom.production.min.js:247)
    at jj (react-dom.production.min.js:216)
    at Z (react-dom.production.min.js:210)
    at $i (react-dom.production.min.js:206)
    at react-dom.production.min.js:114
    at exports.unstable_runWithPriority (scheduler.production.min.js:20)
    at vf (react-dom.production.min.js:113)

**Comment by benradey** [comment]
@xdavidson I made a new commit that fixes the locale format, please re-review

File: pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlUuidProvider.java [hunk 1]
```diff
@@ -30,10 +30,10 @@ public String generateUrlString(DocumentVariant variant) {
             ProductVersion pv = metadata.productVersion().getReference();
             StringBuilder sb = new StringBuilder(getHost(variant.getResourceResolver()));
             sb.append(URL_PREFIX)
-                    .append(getLocale(variant)).append("/")
-                    .append(getDocumentType(variant)).append("/")
+                    .append(getLocale(variant).replaceAll("_", "-").toLowerCase()).append("/") // turns en_US into en-us which is likely a customer portal requirement (need to confirm)
```
Comment (xdavidson) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlUuidProvider.java:33]: we have a method in ServletUtils that transforms the locale to language tag. see
https://github.com/redhataccess/pantheon/blob/master/pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/VariantJsonServlet.java#L109

**Review by xdavidson — APPROVED** [review]
LGTM

**Pull request merged**